spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/RDDFunctions.scala

```diff
@@ -34,4 +38,28 @@ class RDDFunctions[T](rdd: RDD[T]) extends WritableToCassandra[T] with Serializa
   def spanBy[U](f: (T) => U): RDD[(U, Iterable[T])] =
     new SpannedRDD[U, T](rdd, f)
 
+  /**
+   * Uses the data from `RDD` to get data from a Cassandra table. This function is slightly more liberal than
+   * saveToCassandra as you are not required to specify the entire Primary Key. You must specify
+   * the entire partition key but clustering columns are optional. If clustering columns are included they must be in
+   * an order valid for Cassandra.
+   *
+   * The flag repartition (default: true) tells Spark whether or not it should shuffle and repartition the data
+   * so that the queries are preformed on a coordinator for the data they request. When preforming inner-joins between
+   * Cassandra tables this is not neccessary so set this to false so no data shuffling occurs.
+   */
+  def fetchFromCassandra[R](keyspaceName: String, tableName: String, repartition: Boolean = true, partitionsPerReplicaSet:Int = 10)
+                           (implicit connector: CassandraConnector = CassandraConnector(sparkContext.getConf),
+                            newType: ClassTag[R], rrf: RowReaderFactory[R], ev: ValidRDDType[R],
+                            currentType: ClassTag[T], rwf: RowWriterFactory[T]): CassandraRDD[R] = {
```

Comment:
we could avoid the requirement for `currentType: ClassTag[T]` because this class tag is already in rdd wrapped by this `RDDFunctions`. The way to obtain it is `rdd.toJavaRDD().classTag`.